ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java

```diff
@@ -589,6 +600,36 @@ public static String getOperationOrBlobIdFromUri(RestRequest restRequest, RestUt
     return subResource;
   }
 
+  /**
+   * When validating signed path is required by container, this method parses the signed path from uri.
+   * Getting signed path is based on the assumption that the first non-empty segment (split by "/") is signed path.
+   * @param restRequest {@link RestRequest} containing metadata about the request.
+   * @return signed path if uri contains one; otherwise empty string is returned.
+   */
+  public static String getSignedPath(RestRequest restRequest) {
+    String path = restRequest.getPath();
+    SubResource subResource = getBlobSubResource(restRequest);
+    int endIndex = path.length();
+    String signedPath = "";
+    if (subResource != null) {
+      // remove sub resource from path
+      endIndex = endIndex - subResource.name().length() - 1;
+      path = path.substring(0, endIndex);
+    }
+    if (path.startsWith("/")) {
+      // remove leading slash
+      path = path.substring(1);
+    }
+    String[] segments = path.split("/");
+    // extract signed path
+    // 1. if there are more than 1 segment   (i.e. "signed-path/blobId")
+    // 2. if there is 1 segment and uri ends with "/" (i.e. "signed-path/")
+    if (segments.length > 1 || (segments.length == 1 && path.endsWith("/"))) {
```

Comment:
I don't know if it's safe to extract signed path when there is 1 segment and URI ends with "/", i.e. `signed-path/`. Please be strict on this if condition and correct me if any potential counter examples making the code vulnerable.